File: src/main/java/org/labun/jooq/generator/util/NameCreator.java
package org.labun.jooq.generator.util;

import org.jooq.util.Definition;
import org.labun.jooq.generator.config.CodeGenerationConfig;

/**
 * @author Konstantin Labun
 */
public interface NameCreator {
    String createClassName(CodeGenerationConfig cfg, Definition table);

    String createFieldName(CodeGenerationConfig cfg, Definition column);

    String createGetterName(CodeGenerationConfig cfg, Definition column);

    String createSetterName(CodeGenerationConfig cfg, Definition column);

}

File: src/main/kotlin/org/labun/jooq/generator/util/NameCreator.kt
package org.labun.jooq.generator.util

import org.jooq.util.Definition
import org.labun.jooq.generator.config.SubGeneratorConfig

/**
 * @author Konstantin Labun
 */
interface NameCreator {

  fun createClassName(cfg: SubGeneratorConfig, table: Definition): String

  fun createFieldName(cfg: SubGeneratorConfig, column: Definition): String

  fun createGetterName(cfg: SubGeneratorConfig, column: Definition): String

  fun createSetterName(cfg: SubGeneratorConfig, column: Definition): String

}
